core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -152,13 +156,19 @@ class KafkaApis(val requestChannel: RequestChannel,
     // ensureTopicExists is only for client facing requests
     // We can't have the ensureTopicExists check here since the controller sends it as an advisory to all brokers so they
     // stop serving data to clients for the topic being deleted
-    val stopReplicaRequest = request.requestObj.asInstanceOf[StopReplicaRequest]
+    val stopReplicaRequest = request.body.asInstanceOf[StopReplicaRequest]
 
-    authorizeClusterAction(request)
```

Comment:
@ijuma Yeah, I agree with that. I am ok with changing it back and addressing it later with a generic handling function. I agree readability is very important. I have a few things in mind to improve readably and reduce verbosity of Kafka APIs once we migrate most of the requests/responses.